src/itype.h

```diff
@@ -255,6 +255,20 @@ struct islot_book {
     recipe_list_t recipes;
 };
 
+struct islot_mod {
+    /** If non-empty restrict mod to items with those base (before modifiers) ammo types */
+    std::set<ammotype> acceptable_ammo;
+
+    /** If set modifies parent ammo to this type */
+    ammotype ammo_modifier = NULL_ID;
```

Comment:
`NULL_ID` must never be used in context that can be static.
Static initialization is not guaranteed to happen in any order other than those specified for single file.

For example, on Cygwin (just an example), the sort order of files seems totally scrambled.
I think it's the link order which determines static initialization order. There are ways to guarantee static initialization order, but they most likely depend on very specific linking which would be a pain to keep.

Would probably be better to just drop the static `NULL_ID` and wrap it in a getter. We had a PR like this and I think I stalled it, but I can't think of a better way to do it.
We could keep the null id type for that weird overload trick, but it should no longer point to possibly-valid object.